tests/integration_tests/test_hyperopt_ray.py

```diff
@@ -125,6 +130,44 @@ def ray_cluster_4cpu():
         yield
 
 
+class HyperoptTestCallback(TuneCallback):
+    def __init__(self, exp_name: str, model_type: str):
+        self.exp_name = exp_name
+        self.model_type = model_type
+        self.trial_ids = set()
+        self.trial_status = {}
+        self.user_config = {}
+        self.rendered_config = {}
+
+    def on_trial_start(self, iteration: int, trials: List["Trial"], trial: "Trial", **info):  # noqa
+        super().on_trial_start(iteration, trials, trial, **info)
+        self.trial_ids.add(trial.trial_id)
+
+    def on_trial_complete(self, iteration: int, trials: List["Trial"], trial: "Trial", **info):  # noqa
+        super().on_trial_complete(iteration, trials, trial, **info)
+        self.trial_status[trial.trial_id] = trial.status
+
+        model_hyperparameters = os.path.join(
+            trial.logdir, f"{self.exp_name}_{self.model_type}", "model", MODEL_HYPERPARAMETERS_FILE_NAME
+        )
+        if os.path.isfile(model_hyperparameters):
+            try:
+                with open(model_hyperparameters) as f:
+                    json.load(f)
```

Comment:
Is it worth adding a check here to make sure the loaded json isn't empty?

Reply:
Same for the json load below

Reply:
Sure, I can add that.